src/hooks/useFriction.ts

```diff
@@ -28,26 +30,29 @@ export const useFriction = <M extends HTMLElement | SVGElement = any>({
 
   const { controller } = React.useMemo(() => {
     const interpolators = getInterpolatorsForPairs({ from, to });
+    const maxPosition = getMaxDistanceFriction({
+      mu: config.mu,
+      initialVelocity: config.initialVelocity,
+    });
 
     return friction1D({
       config,
       onUpdate: ({ position }) => {
         interpolators.forEach(({ interpolator, property, values }) => {
           const value = interpolator({
-            range: [
-              0,
-              getMaxDistanceFriction({
-                mu: config.mu,
-                initialVelocity: config.initialVelocity,
-              }),
-            ],
+            range: [0, maxPosition],
             domain: [values.from, values.to],
             value: position[0],
           });
 
           if (ref.current) {
             ref.current.style[property as any] = `${value}`;
           }
+
+          if (onFrame) {
+            const progress = position[0] / maxPosition;
+            onFrame(progress);
+          }
```

Comment:
Here's where we'll execute the `onFrame` logic if provided. This feels a bit cleaner than returning `progress` as a third part of the tuple returned by this hook. I'm also not sure we could really pull that off. The parent component would have to re-render to see the updated value of a progress `ref`, which would restart the animation 😬